libcpuid/libcpuid_types.h

```diff
@@ -58,8 +58,12 @@ typedef unsigned short		uint16_t;
 	typedef unsigned __int64	uint64_t;
 #else
 	/* all other sane compilers: */
-	typedef signed long long   int64_t;
-	typedef unsigned long long uint64_t;
+	#if __WORDSIZE == 64
+	typedef unsigned long int   uint64_t;
+	#else
+	__extension__
```

Comment:
Isn't `__extension__` gcc-only?

Reply:
yes.

Reply:
my system:
bash-4.4# uname -a
Linux hostname 4.14.14 #2 SMP Wed Jan 17 12:56:07 CST 2018 x86_64 Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz GenuineIntel GNU/Linux
bash-4.4# gcc -v
gcc version 7.2.0 (GCC)

Reply:
Your system should have `<stdint.h>`, but for some reason it appears it doesn't (if it had, you wouldn't take the `#else` branch of the `#if defined(HAVE_STDINT_H)`).

Can you please check why does it seem your machine doesn't have stdint.h? For starters, how are you invoking the compiler, are you using `./configure`? If it's a custom compile process, adding `-DHAVE_STDINT_H` to your gcc command line should fix things without any changes to `libcpuid_types.h`.

Reply:
I have this same issue as well, I have no idea why, on `gcc version 6.3.0 20170519 (Ubuntu/Linaro 6.3.0-18ubuntu2~16.04) `

Reply:
I can confirm that this PR solved the problem on my machine as long as I include `libcpuid.h` after `inttypes.h`.

Reply:
@ferdna, @praisethemoon I cannot reproduce your issue. I tried the following:

1. Installed gcc-6 on a stock Ubuntu Server machine. My `gcc -v` gives `gcc version 6.3.0 20170519 (Ubuntu/Linaro 6.3.0-18ubuntu2~16.04)`, exactly what @praisethemoon has;
2. I downloaded the latest libcpuid from github;
3. I ran `libtoolize` and `autoreconf --install` as per the Readme
4. I ran `./configure`, `make`, all compiled fine
5. `make test` works without issues
6. `make install` followed by writing a small program showed no issues. Program code:

```#include <libcpuid.h>
#include <stdio.h>

int main(void)
{
	struct cpu_id_t t;
	cpu_identify(NULL, &t);
	printf("Codename: %s\n", t.cpu_codename);
	return 0;
}
```
This compiles fine from the command line and works as expected.

Any thoughts? I wonder what may be different in your setup.

Reply:
@anrieff I have used the v0.4.0 release which seems to have the same header file as the current repo changes. Thanks for looking into this!

Reply:
I tried with 0.4.0 on that gcc 6, it compiles fine as well.
Can you paste what errors do you encounter when you try to compile the stock libcpuid code (e.g. 0.4.0) without the PR modifications?